clients/roscpp/src/libros/param.cpp

```diff
@@ -720,6 +720,31 @@ bool getCached(const std::string& key, std::map<std::string, bool>& map)
   return getImpl(key, map, true);
 }
 
+bool getParamNames(std::vector<std::string>& keys)
+{
+  XmlRpc::XmlRpcValue params, result, payload;
+  params[0] = this_node::getName();
+  if (!master::execute("getParamNames", params, result, payload, false))
+    return false;
+  // Make sure it's an array type
+  if (result.getType() != XmlRpc::XmlRpcValue::TypeArray)
+    return false;
+  // Make sure it returned 3 elements
+  if (result.size() != 3)
+    return false;
+  // Get the actual parameter keys
+  XmlRpc::XmlRpcValue parameters = result[2];
+  // Resize the output
+  keys.resize(parameters.size());
+
+  // Fill the output vector with the answer
+  for (int i = 0; i < parameters.size(); i++) {
```

Comment:
Thanks for pointing it out. I obviously didn't check the size but expected it to be `size_t`. I will change this back as part of PR #760.